### PR TITLE
Add flow types for browserslist

### PR DESCRIPTION
**Pull request by thebriando**
# ↪️ Pull Request
Adds a flow type lib def for `browserslist`.

## 🚨 Test instructions

Run `yarn flow` to ensure that there are no flow errors.

@padmaia @wbinnssmith

### CODE REVIEW

File: flow-libs/browserslist.js.flow [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// @flow strict-local
+
+declare module 'browserslist' {
+  import type {ConfigResult} from '@parcel/types';
+
+  declare type BrowserslistConfig = {|[string]: string | Array<string>|};
+
+  declare module.exports: {|
+    (browsers: string | Array<string>): Array<string>,
+    parseConfig(configResult: ConfigResult): Promise<?BrowserslistConfig>,
```
Comment (wbinnssmith) [inline flow-libs/browserslist.js.flow:10]: looks like this takes a string: https://github.com/browserslist/browserslist/blob/88f7d5141f9be0ac3d712f8b52f4654fadbdeeff/node.js#L260

File: flow-libs/browserslist.js.flow [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    loadConfig({[path: FilePath]: FilePath, ...}): string,
```
Comment (wbinnssmith) [inline flow-libs/browserslist.js.flow:11]: not sure how `FilePath` works here. A string is probably better, or we could define `FilePath` inside this module as an alias to `string`.

Comment (wbinnssmith) [inline flow-libs/browserslist.js.flow:11]: the first argument, options, is defined in browserlist's js api docs: https://github.com/browserslist/browserslist#js-api

File: packages/core/core/src/public/Environment.js [hunk 3]
```diff
@@ -86,6 +86,7 @@ export default class Environment implements IEnvironment {
     return this;
   }
 
+  // flowlint unsafe-getters-setters: off
```
Comment (wbinnssmith) [inline packages/core/core/src/public/Environment.js:89]: this should probably be hoisted to the top of the file

File: packages/transformers/babel/src/getBabelTargets.js [hunk 4]
```diff
@@ -66,6 +66,7 @@ export default async function getBabelTargets(
       browsers != null &&
       !Array.isArray(browsers)
     ) {
+      // flowlint sketchy-null-string: off
       let env = process.env.NODE_ENV || 'development';
```
Comment (wbinnssmith) [inline packages/transformers/babel/src/getBabelTargets.js:70]: @padmaia @devongovett looks like this is using the real environment's `NODE_ENV`. 

@thebriando `process.env.NODE_ENV ?? 'development` addresses this, though it changes the semantics, probably for the better. In the future, we can avoid disabling lint by making the falsy check explicit like `(foo !== '' && foo != null) ? foo : 'fallback'`

File: packages/transformers/babel/src/getBabelTargets.js [hunk 5]
```diff
@@ -66,11 +66,11 @@ export default async function getBabelTargets(
       browsers != null &&
       !Array.isArray(browsers)
     ) {
-      let env = process.env.NODE_ENV || 'development';
+      let env = process.env.NODE_ENV ?? 'development';
       browsers = browsers[env] || browsers.defaults;
     }
 
-    if (browsers) {
+    if (browsers !== null && browsers !== undefined) {
```
Comment (wbinnssmith) [inline packages/transformers/babel/src/getBabelTargets.js:73]:
```suggestion
    if (browsers != null) {
```

File: flow-libs/browserslist.js.flow [hunk 6]
```diff
@@ -0,0 +1,21 @@
+// @flow strict-local
+
+declare module 'browserslist' {
+  declare type BrowserslistConfig = {|[string]: string | Array<string>|};
+  declare type Options = {
```
Comment (wbinnssmith) [inline flow-libs/browserslist.js.flow:5]: can this be exact?

File: packages/core/core/src/public/Environment.js [hunk 7]
```diff
@@ -1,4 +1,5 @@
-// @flow
+// @flow strict-local
+// flowlint unsafe-getters-setters: off
```
Comment (wbinnssmith) [inline packages/core/core/src/public/Environment.js:2]: @devongovett @mischnic what do you think of allowing getters and setters globally? They do technically introduce potential unsoundness.

Comment (mischnic) [reply packages/core/core/src/public/Environment.js:2]: 👍  We do use them extensively in the public interfaces. And I think as long as we keep them side-effect-free (which no linter can verify anyway), it should be fine regarding soundness.

File: flow-libs/browserslist.js.flow [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    config?: string,
+    ignoreUnknownVersions?: boolean,
+    dangerousExtend?: boolean,
+    mobileToDesktop?: boolean,
+  |};
+
+  declare module.exports: {|
+    (query?: string | Array<string>, opts?: Options): Array<string>,
+    parseConfig(configResult: string): BrowserslistConfig,
+    loadConfig(opts: {[path: string]: string, ...}): string,
```
Comment (wbinnssmith) [inline flow-libs/browserslist.js.flow:18]: can this be exact?

**Review by wbinnssmith — APPROVED** [review]

**Pull request merged**